### PR TITLE
Fix: Update Fine-tuning Tutorial and Structure

### PR DESCRIPTION
**Pull request by Shuang-cnt**
# Description

Update the Full fine-tuning doc
 - add Pre-requisite section
 - add Set environment variables section
 - reorganize structure 

# Tests

Verified the updated documentation by walking through the entire workflow, including environment setup and workload submission.
Attached logs:
[Convert Meta to Maxtext](https://paste.googleplex.com/6282563092086784)
[Download C4 Dataset](https://paste.googleplex.com/5483730850414592)
[Covert Maxtext to HF](https://paste.googleplex.com/6077315698262016)
[Sample Training](https://paste.googleplex.com/6371804593651712)

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: docs/tutorials/posttraining/full_finetuning.md
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/full_finetuning.md:18]: Can we add a short paragraph explaining what Full Fine-Tuning is, similar to SFT/RL docs?

File: docs/tutorials/posttraining/full_finetuning.md [hunk 1]
```diff
@@ -31,21 +31,44 @@ The high level steps involve:
...
 - Running the training script with the checkpoint
-- Note: You may need to change the training parameters to fit the model to the
-  TPU or GPU shape and to obtain an optimized performance.
+- Note: Training parameters may require adjustment to align the model with the specific TPU or GPU topology and achieve optimal performance.
 
 ## MaxText checkpoints
 
 MaxText checkpoints are in their own format. You can see the format in the script for llama conversion script.
 
+### Meta's PyTorch checkpoint to Maxtext (Orbax) checkpoint
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/full_finetuning.md]: Can we update this to use the new checkpoint conversion script?
Reference: https://maxtext.readthedocs.io/en/latest/tutorials/posttraining/sft.html#option-2-converting-a-hugging-face-checkpoint

File: docs/tutorials/posttraining/full_finetuning.md
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/full_finetuning.md]: nit: missing `.` at the end of the statement. Please fix this in the entire doc.

Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/full_finetuning.md:1]: Can we format this doc similar to other post-training docs?

Reference: https://maxtext.readthedocs.io/en/latest/tutorials/posttraining/sft.html

Comment (Shuang-cnt) [reply docs/tutorials/posttraining/full_finetuning.md:18]: Added. Please check. Thanks

File: docs/tutorials/posttraining/full_finetuning.md (hunk 1 above)
Comment (Shuang-cnt) [reply docs/tutorials/posttraining/full_finetuning.md]: update both Maxtext to HF and HF to Maxtext script to the latest (to_maxtext.py and to_huggingface.py)

File: docs/tutorials/posttraining/full_finetuning.md
Comment (Shuang-cnt) [reply docs/tutorials/posttraining/full_finetuning.md]: updated. Thanks

Comment (Shuang-cnt) [reply docs/tutorials/posttraining/full_finetuning.md:1]: updated, thanks

File: docs/tutorials/posttraining/full_finetuning.md [hunk 2]
```diff
@@ -14,88 +14,126 @@
...
+python3 -m pip install torch --index-url https://download.pytorch.org/whl/cpu  # Ensure torch is installed for the conversion script
 
-#### Dataset
+python3 -m MaxText.utils.ckpt_conversion.to_maxtext src/MaxText/configs/base.yml \
+    model_name=${MODEL_NAME} \
+    hf_access_token=${HF_TOKEN} \
+    base_output_directory=${MODEL_CKPT_DIRECTORY} \
+    scan_layers=True skip_jax_distributed_system=True
+```
+## MaxText checkpoint to Hugging Face checkpoint
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/full_finetuning.md]: This might be irrelevant to this documentation. Please remove.

File: docs/tutorials/posttraining/full_finetuning.md [hunk 3]
```diff
@@ -14,88 +14,126 @@
...
+
+Below is a sample training script for LLama2-7b on v6e-8 TPU VM.
+
+```sh
+python3 -m MaxText.train \
+  src/MaxText/configs/base.yml \
+  run_name="llama2-finetune-maxtext" \
+  base_output_directory=${BASE_OUTPUT_DIRECTORY} \
+  load_parameters_path=${MODEL_CKPT_PATH} \
+  model_name='llama2-7b' \
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/full_finetuning.md]: nit: `${MODEL_NAME}`

File: docs/tutorials/posttraining/full_finetuning.md [hunk 4]
```diff
@@ -14,88 +14,126 @@
...
+The above will download the c4 dataset to the GCS BUCKET.
+
+## Sample Full Fine tuning script
+
+Below is a sample training script for LLama2-7b on v6e-8 TPU VM.
+
+```sh
+python3 -m MaxText.train \
+  src/MaxText/configs/base.yml \
+  run_name="llama2-finetune-maxtext" \
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/full_finetuning.md]: Can be moved to `Setup environment variables` section?

File: docs/tutorials/posttraining/full_finetuning.md [hunk 5]
```diff
@@ -14,88 +14,126 @@
...
+  base_output_directory=${BASE_OUTPUT_DIRECTORY} \
+  load_parameters_path=${MODEL_CKPT_PATH} \
+  model_name='llama2-7b' \
+  dataset_path=${DATASET_GCS_BUCKET} \
+  async_checkpointing=False  \
+  steps=10 per_device_batch_size=1
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/full_finetuning.md]: We might need to set `tokenizer_path` and `hf_access_token`, right?

File: docs/tutorials/posttraining/full_finetuning.md [hunk 6]
```diff
@@ -14,88 +14,126 @@
...
+## Sample Full Fine tuning script
+
+Below is a sample training script for LLama2-7b on v6e-8 TPU VM.
+
+```sh
+python3 -m MaxText.train \
+  src/MaxText/configs/base.yml \
+  run_name="llama2-finetune-maxtext" \
+  base_output_directory=${BASE_OUTPUT_DIRECTORY} \
+  load_parameters_path=${MODEL_CKPT_PATH} \
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/full_finetuning.md:123]: `MODEL_CKPT_PATH` won't be set if user follows `Option 2: Converting a Hugging Face checkpoint`. Please update the documentation.

File: docs/tutorials/posttraining/full_finetuning.md (hunk 2 above)
Comment (Shuang-cnt) [reply docs/tutorials/posttraining/full_finetuning.md]: removed,Thanks

File: docs/tutorials/posttraining/full_finetuning.md (hunk 3 above)
Comment (Shuang-cnt) [reply docs/tutorials/posttraining/full_finetuning.md]: set it in the `Setup environment variables` section

File: docs/tutorials/posttraining/full_finetuning.md (hunk 4 above)
Comment (Shuang-cnt) [reply docs/tutorials/posttraining/full_finetuning.md]: set it in the `Setup environment variables` section

File: docs/tutorials/posttraining/full_finetuning.md (hunk 5 above)
Comment (Shuang-cnt) [reply docs/tutorials/posttraining/full_finetuning.md]: set it in the `Setup environment variables` section and added them in the command

File: docs/tutorials/posttraining/full_finetuning.md (hunk 6 above)
Comment (Shuang-cnt) [reply docs/tutorials/posttraining/full_finetuning.md:123]: add 1 sample for option 1 and 1 sample for option 2

Comment (SurbhiJainUSC) [reply docs/tutorials/posttraining/full_finetuning.md:123]: Instead of duplicating the training command, you can just set `MODEL_CKPT_PATH` in Option 2 after checkpoint is converted.

Comment (Shuang-cnt) [reply docs/tutorials/posttraining/full_finetuning.md:123]: @SurbhiJainUSC  thanks for the point!
add `export MODEL_CKPT_PATH=${MODEL_CKPT_DIRECTORY}/0/items` in Option 2
